### ISSUE TITLE
Disable Google Analytics reporting by default

### ISSUE THREAD

**Issue by ava1ar** [reportportal/client-java#125]
Hi guys.

With recent ReportPortal client libraries update we noticed following:
```
2020-09-18 10:43:08 [pool-3-thread-1] ERROR c.e.r.s.analytics.GoogleAnalytics - Connect to www.google-analytics.com:443 [www.google-analytics.com/172.217.7.14] failed: Connection timed out: connect
```

Looks like GoogleAnalytics reporting was silently added and activated by default? This is pretty bad move, since lots of companies are using ReportPortal are in finance sector, where such "integration" is violating company infosec policies.
Also, when it can't reach the google servers (since the corporate proxy), it adds few seconds timeout at the end of execution, which is annoying. 

Can you please disable this integration by default or at least provide a configuration option to turn it off. Also, would be nice to see information about such features in the documentation. Thanks!

**Comment by evjlobanova** [comment]
Hello!
Information about this feature is added to all GitHub repositories in the Read Me file. If you find a repository without a disclaimer, please send us this information. We will update it accordingly.

For now, we do not have plans to turn off GA for agents since we should support more than 10 different test frameworks (+ different TF versions) and at the same time we have a limited team capacity.

In this situation, the only option to define the most popular TFs in order to support them is to use GA for agents.

At the same time in RP app application, you can swith GA On/Off.

One more important thing is that GA in case of reporting does not collect any personal data, only agent name, and version.

**Comment by ava1ar** [comment]
Hi,

Thanks for the response! Can you at least add a configuration option to turn it off? Out test setup doesn't have access to the internet and attempt to report to GA fails, but each time there is a delay before test is completed. What we supposed to do in this case?

**Comment by evjlobanova** [comment]
As I said for now we do not have such intention because of reasons higher.

@KatsiarynaTatarynovich do you have any ideas on how the problem with delay can be resolved?

**Comment by ava1ar** [comment]
> As I said for now we do not have such intention because of reasons higher.

I understand your requirements, but I also limited in options and can't enable internet access for scripts. From what I see from the source code, GA reporting uses same timeout value which is configured for ReportPortal itself, so I can't put it to very low value to avoid delays (I may get timeouts for the RP reporting). Still looking for proper resolution from your side, but it looks like we will need to fork the extension on our end and remove GA integration this way.

**Comment by HardNorth** [comment]
@ava1ar as @evjlobanova said, we need this information for our long term planning. But I don't want everyone remove GA reporting just because of their fears despite the fact that the code is open and it easy to check what we send. Feel free to update it by yourself, or you can simply configure analytics domain to point on local host.

**Comment by ava1ar** [comment]
@HardNorth looks like you never worked in financial domain and only take into account technical side of the things...
The concern is not only the fact it is GA. Issue is that to make it work I need to configure corporate proxy for tests (just for this purpose, normally I don't want my test to have internet access). And `you can simply configure analytics domain to point on local host` is not that simple, when you don't have root/admin access to the machines you are working on.
I definitely can work-around this, but approach you are taking here is not constructive unfortunately :(

**Comment by evjlobanova** [comment]
@ava1ar Vadzim has extended expertise in different domains.  But also it should be taken into account that ReportPortal is an open-source tool, that we should consider the interests of different areas and different customers. 

Thank you for highlighting this situation. As I told, we would discuss your case, and get back to you as soon as we make a decision on how to deal with this tricky case. I can not tell you exactly estimations, because the team has limited capacity.

But as Vadzim pointed,  ReportPortal.io is opensource, feel free to send us a pull request with your changes or try to update/ or create our agent with your own improvements.

**Comment by ava1ar** [comment]
> As I told, we would discuss your case, and get back to you as soon as we make a decision on how to deal with this tricky case.

Thanks, would be nice to get some kind of solution for this.

Sorry if I misinterpreted your response, I saw issues was closed, which usually means no actions to be taken. Glad if this is not the case in this situation.

**Comment by evjlobanova** [comment]
We have a meeting on this question this week.
And after that, I update this ticket. Feel free to ping me

**Comment by HardNorth** [comment]
@evjlobanova OK reopening this for now.

**Comment by HardNorth** [comment]
We leave it as is.

**Comment by ava1ar** [comment]
No workaround for timeouts when internet is not available?

**Comment by HardNorth** [comment]
Not from our side, you can ask your devops for the workaround I already suggested.

**Comment by HardNorth** [comment]
@ava1ar After few more internal discussions we decided to rework our analytics publishing class and add an environment variable `AGENT_NO_ANALYTICS`. It will be available in the next client releases (post 5.0.15). To use it just set something into `AGENT_NO_ANALYTICS` environment variable.

**Comment by ava1ar** [comment]
Thanks! This should great and should work just fine! Will check it out once available.
